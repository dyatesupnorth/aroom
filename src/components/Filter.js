import React, { Component } from "react";
import { connect } from "react-redux";
import {
  setTextFilter,
  setStarRatingFilter,
  setFacilitiesFilter
} from "../actions/filters";

export class Filter extends Component {
  
  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  };

  onStarRatingChange = e => {
    this.props.setStarRatingFilter(e.target.value);
  };

  onFacilitiesChange = (e, i) => {
    this.props.setFacilitiesFilter(e.target.value);
  };

  renderRadioButtons(maxStarRating) {
    // TODO: Pull max star rating dervied from max of all hotels
    // TODO: Convert to checkboxes, we want to be able to filter based on multiple selections
    let radioButtons = [];
    for (let i = 0; i < maxStarRating; i++) {
      radioButtons.push(
        <span key={i}>
          <input
            type="radio"
            value={i + 1}
            name="starRating"
            onChange={this.onStarRatingChange}
          />
          {i + 1}
        </span>
      );
    }
    return radioButtons;
  }

  renderCheckBoxes() {
    // TODO: Pull all available facilities directly from available hotels.
    return this.props.availableFacilities.map((facility, i) => (
      <label key={i}>
        {facility}
        <input
          name={facility}
          value={facility}
          type="checkbox"
          onChange={this.onFacilitiesChange}
        />
      </label>
    ));
  }

  render() {
    return (
      this.props.filters && (
        <div style={{ border: "1px solid tomato" }}>
          <h1>Filter</h1>
          <div className="form-group">
            <label>Name</label>
            <input
              className="form-control"
              name="text"
              type="text"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="form-group">
            <label>Star Rating</label>
            {this.renderRadioButtons(5)}
          </div>
          <div className="form-group">
            <label>Facilities</label>
            {this.renderCheckBoxes()}
          </div>
        </div>
      )
    );
  }
}

const mapStateToProps = state => {
  console.log("â€‹state", state);
  return {
    filters: state.filters,
    // Trying to get too fancy!
    availableFacilities: state.hotels.reduce(
      (acc, hotel) => (
        [].push.apply(acc, hotel.facilities.filter(v => acc.indexOf(v) === -1)),
        acc
      ),
      []
    )
  };
};

const mapDispatchToProps = dispatch => {
  return {
    setTextFilter: text => dispatch(setTextFilter(text)),
    setStarRatingFilter: starRating =>
      dispatch(setStarRatingFilter(starRating)),
    setFacilitiesFilter: facility => dispatch(setFacilitiesFilter(facility))
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Filter);
