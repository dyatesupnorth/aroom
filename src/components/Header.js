import React, { Component } from "react";

export class Header extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Header</h1>
      </div>
    );
  }
}

export default Header;
