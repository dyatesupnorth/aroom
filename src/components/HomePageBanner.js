import React, { Component } from "react";

export class HomePageBanner extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>HomePageBanner</h1>
      </div>
    );
  }
}

export default HomePageBanner;
