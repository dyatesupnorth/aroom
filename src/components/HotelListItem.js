import React from "react";
import ListGroup from "react-bootstrap/ListGroup";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar } from "@fortawesome/free-solid-svg-icons";
export const renderStars = noOfStars => {
	console.log("​noOfStars", noOfStars)
  let stars = [];
	
  for (let i = 0; i < noOfStars; i++) {
		console.log("​i", i)
    stars.push(<FontAwesomeIcon icon={faStar} />);
  }
  console.log("​stars", stars)
  return stars;
};
export const HotelListItem = props => {
  return (
    <ListGroup.Item className="list-group-item-action flex-column align-items-start">
      <div className="d-flex w-100 justify-content-between">
        <h5 className="mb-1">{props.name}</h5>
      </div>
      <p className="mb-1">{renderStars(props.starRating)}</p>
      <small>
        {!props.facilities || props.facilities.length === 0 ? (
          ''
        ) : (
          <span>
            <ul className="list-inline">
              {props.facilities.map(facility => {
                return (
                  <li className="list-inline-item" key={facility}>
                    {facility}
                  </li>
                );
              })}
            </ul>
          </span>
        )}
      </small>
    </ListGroup.Item>
  );
};

export default HotelListItem;
