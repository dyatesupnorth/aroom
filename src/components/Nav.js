import React, { Component } from "react";

export class Nav extends Component {
  render() {
    return (
      <div style={{ border: "1px solid tomato" }}>
        <h1>Nav</h1>
      </div>
    );
  }
}

export default Nav;
