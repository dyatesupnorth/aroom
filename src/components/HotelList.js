import React from "react";
import { connect } from "react-redux";
import selectHotels from "../selectors/hotels";
import { HotelListItem } from "./HotelListItem";
import Card from "react-bootstrap/Card";

import ListGroup from "react-bootstrap/ListGroup";

export const HotelList = props => {
  return (
    <Card>
      <ListGroup variant="flush">
        {props.hotels.length === 0 ? (
          <p>No hotels</p>
        ) : (
          props.hotels.map(hotel => {
            return <HotelListItem key={hotel.name} {...hotel} />;
          })
        )}
      </ListGroup>
    </Card>
  );
};

const mapStateToProps = state => {
  return {
    hotels: selectHotels(state.hotels, state.filters)
  };
};
export default connect(
  mapStateToProps,
  undefined
)(HotelList);
