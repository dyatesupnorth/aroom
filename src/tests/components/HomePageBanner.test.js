import React from "react";
import { shallow } from "enzyme";
import { HomePageBanner } from "../../components/HomePageBanner";

test("should render HomePageBanner", () => {
  const wrapper = shallow(<HomePageBanner />);
  expect(wrapper).toMatchSnapshot();
});
