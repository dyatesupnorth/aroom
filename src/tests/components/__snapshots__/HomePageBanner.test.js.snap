// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomePageBanner 1`] = `
<Bootstrap(Jumbotron)
  fluid={true}
  style={
    Object {
      "backgroundImage": "url(https://images.unsplash.com/photo-1496417263034-38ec4f0b665a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1051&q=80)",
      "backgroundPosition": "center",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
      "minHeight": "30rem",
    }
  }
>
  <h2
    className="mt-3 text-white"
  >
    Welcome to ARoom
  </h2>
</Bootstrap(Jumbotron)>
`;
